src/anndata/experimental/backed/_compat.py

```diff
@@ -0,0 +1,42 @@
+from __future__ import annotations
+
+try:
+    from xarray import DataArray
+except ImportError:
+
+    class DataArray:
+        def __repr__(self) -> str:
+            return "mock DataArray"
+
+
+try:
+    import xarray as xr
+except ImportError:
+    xr = None
+
+
+try:
+    from xarray.backends.zarr import ZarrArrayWrapper
+except ImportError:
+
+    class ZarrArrayWrapper:
+        def __repr__(self) -> str:
+            return "mock ZarrArrayWrapper"
```

Comment:
This code means that you get a really confusing error if you run `read_backed` without `xarray` being installed

Reply:
Yes but unfortunately I think this compat layer is needed so that core is compatible.  Maybe we can add a check within `read_backed`?